src/views/WorkspaceDetail.vue

```diff
@@ -211,31 +211,14 @@ export default Vue.extend({
       }
     },
     async update(this: any) {
-      // Get lists of node and edge tables.
-      let nodeTables;
-      let edgeTables;
-
-      try {
-        this.loading = true;
-        nodeTables = await api.tables(this.workspace, { type: 'node' });
-        edgeTables = await api.tables(this.workspace, { type: 'edge' });
-      } catch (err) {
-        this.loading = false;
-        if (err.status === 404 && err.statusText === 'Workspace Not Found') {
-          this.$router.replace({name: 'home'});
-        } else {
-          throw err;
-        }
+      this.loading = true;
 
-        return;
+      await store.dispatch.fetchWorkspace(this.workspace);
+      if (Object.keys(this.$refs).length) {
```

Comment:
Why is this guard needed?

Reply:
I think because we're calling update on the `created` lifecycle hook, this can run before the refs are actually available.

Reply:
Ok.